packages/legend-extension-dsl-service/src/components/studio/ServiceRegisterModal.tsx

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+import { observer } from 'mobx-react-lite';
+import {
+  ActionState,
+  assertErrorThrown,
+  assertNonEmptyString,
+  guaranteeNonNullable,
+  LogEvent,
+  uuid,
+} from '@finos/legend-shared';
+import {
+  generateServiceManagementUrl,
+  LATEST_PROJECT_REVISION,
+  LEGEND_STUDIO_APP_EVENT,
+} from '@finos/legend-application-studio';
+import { createServiceElement } from '../../stores/studio/QueryProductionizerStore.js';
+import {
+  CheckSquareIcon,
+  clsx,
+  CustomSelectorInput,
+  Dialog,
+  Modal,
+  ModalTitle,
+  Panel,
+  PanelFullContent,
+  PanelLoadingIndicator,
+  RocketIcon,
+  SquareIcon,
+} from '@finos/legend-art';
+import {
+  ActionAlertType,
+  ActionAlertActionType,
+} from '@finos/legend-application';
+import {
+  ServiceExecutionMode,
+  isValidFullPath,
+  validate_ServicePattern,
+} from '@finos/legend-graph';
+import type { ExistingQueryEditorStore } from '@finos/legend-application-query';
+import { ProjectData } from '@finos/legend-server-depot';
+
+const ServiceRegisterModal = observer(
+  (props: { editorStore: ExistingQueryEditorStore; onClose(): void }) => {
+    const { editorStore, onClose } = props;
+    const [registrationState] = useState(ActionState.create());
+    const [text, setText] = useState('');
+    const [serviceEnv, setServiceEnv] = useState<string | undefined>(undefined);
```

Comment:
We should avoid defaulting to `null|undefined` is possible.
So please:
- activate service should be defaulted to `true`
- the service env should be defaulted to the first option
- owners should IF POSSIBLE. have the current user as the first owner.